docs/user-guide/configuring.md

```diff
@@ -703,6 +703,65 @@ module.exports = {
 }
 ```
 
+## Glob Pattern based Configuration
+
+Sometimes a more fine-controlled configuration is necessary e.g. if the configuration for files within the same directory has to be different. Therefore you can provide configurations that will only apply to files that match a specific glob pattern (based on [minimatch](https://github.com/isaacs/minimatch)).
```

Comment:
I had a look at using the existing glob code, but it seems to be a different use case (it uses `node-glob`, which is the filesystem-aware version of `minimatch`). In this case, we've already retrieved a list of candidate files from the filesystem, and we just use `minimatch` to filter them. Since `node-glob` uses `minimatch` under the hood, I think the results should be essentially the same, so I don't think there's too much potential for confusion.